AGameRules/AGameRules.cpp

```diff
@@ -51,10 +51,9 @@ void AGameRules::Tick(float deltaTime) {
 UWorld* g_pWorld = NULL;
 void    AGameRules::BeginPlay() {
     Super::BeginPlay();
-    Msg("AGameRules::BeginPlay\n");
-    if (s_iEntityCount != g_entList.Num()) {
-        NLogger::Fatal("\nIBaseEntity::s_iEntityCount != g_entList.Num()");
-    }
+    ADebug::Log("AGameRules::BeginPlay\n");
+    ADebug::Assert((bool)(s_iEntityCount = g_entList.Num()),
+                   "\nIBaseEntity::s_iEntityCount = g_entList.Num()");

```

Comment:
`Assert` already comes with a "built-in" if statement to verify the condition passes